### ISSUE TITLE
Duplicated source code filenames

### ISSUE THREAD

**Issue by jaime-olivares** [Efferent-Health/fo-dicom#2]
Found two sets of source code files in different folder levels. Please explain why is that.

<img width="290" alt="screen shot 2017-01-12 at 22 14 56" src="https://cloud.githubusercontent.com/assets/6194083/21917419/99fdf7ee-d915-11e6-876e-523ac56e239d.png">

**Comment by anders9ustafsson** [comment]
@jaime-olivares Those files are related to the .NET Framework (root level) and *Universal Windows Platform* (*Windows* sub-folder), respectively.

The duplicates are there because .NET Framework and UWP supports two different flavors of C++, C++/CLI (.NET) and C++/CX (UWP). 

Neither implementation will work out-of-the-box in pure C++. but you will need to create a second copy and adapt these to pure C++ to ensure that the files build on Linux.

I recommend that you base your adaptation on the C++/CX implementation, i.e. the files in the *Windows* sub-folder (all of them). This implementation is already prepared for communication with the managed code that does *not* require callbacks.

**Comment by jaime-olivares** [comment]
@anders9ustafsson Got that, but I think two different code sets for different platforms are not really good. that will be ever worse when we try to introduce one or many Linux flavors plus macOS. The traditional way to handle this in cross-platform-C-code is to use #define's and #ifdef directives. Take as an example https://sourcecodebrowser.com/fbb/7.04j/fbb__serv_8h_source.html
The code may look convoluted but any modern editor will show as disabled the code portions that don't meet the current definitions.
So, in summary, I suggest to have a single code base with multiple directives. We can start trying with the existing 2 platforms.

**Comment by anders9ustafsson** [comment]
@jaime-olivares I perfectly agree that multiple code sets are sub-optimal. On the other hand, the C++/CLI and C++/CX implementations differ so much that merging the two code sets would practically mean two distinct sections of /CLI and /CX code, respectively, per file.

It is likely that a P/Invoke implementation could be reused on all platforms, i.e. .NET, Windows Universal Platform, Xamarin and Mono, and eventually I hope that your work could lead to this unification. Then we could lose the C++/CLI and C++/CX flavors without problems.

Therefore, right now I think it is more fruitful if you create a second copy of this code set, where the C++/CX implementation probably is the preferred, adapt it to pure C/C++ and make it accessible to P/Invoke calls. Ideally, such an effort will make it easier to eventually reuse your code set in .NET and UWP and allow for full retirement of the CLI and CX implementation.
